src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -370,10 +300,9 @@ private static bool IsArraySegment(Type t)
             if (type.IsDefined(Globals.TypeOfCollectionDataContractAttribute, false))
                 return false;
 
-            Type[] interfaceTypes = type.GetInterfaces();
-
             if (!IsArraySegment(type))
             {
+                Type[] interfaceTypes = type.GetInterfaces();
                 foreach (Type interfaceType in interfaceTypes)
```

Comment:
I see this is an improvement from .NET Framework. If there any need to use a local variable to hold the interfaces? Could you put the method call inline inside the foreach?